app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -21,6 +21,20 @@ def fetch_middleware_servers
             server = persister.middleware_servers.find_or_build(eap.path)
             parse_middleware_server(eap, server)
 
+            # if immutable or in container flag is not set, check also the agent resource
```

Comment:
I agree and the same holds for the [next 12 lines](https://github.com/Jiri-Kremser/manageiq-providers-hawkular/blob/b4627462ee5bf9824f5989f24c0ad95928991ee9/app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb#L38:L50), but let's not complicate this as this should go in as fast as possible. We can do that in a followup PR.